public/app/plugins/datasource/loki/syntax.ts

```diff
@@ -69,7 +69,7 @@ export const FUNCTIONS = [...AGGREGATION_OPERATORS, ...RANGE_VEC_FUNCTIONS];
 
 const tokenizer: Grammar = {
   comment: {
-    pattern: /(^|[^\n])#.*/,
+    pattern: /(^|[^\n"a-zA-Z])#.*/,
```

Comment:
So I do not think we can solve it by single regex here. This still does not match comment on a new line and if you do `##` inside the label value it still matches as comment.

What we can do is set this to simple `/#.*/` pattern, then make `context-labels` `greedy: true`. This will make it so that `context-labels` will be matched first and so no comment will be matched inside that group. But then we can add the same `comment: /#.*/,` to be in the `inside` key and because `label-value` is also greedy it will now allow comment inside the `label-value`.

Probably need to be tested though is there is no edge case.